test/gtest/mhp/sort.cpp

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+#include <dr/detail/communicator.hpp>
+#include <dr/mhp/algorithms/sort.hpp>
+
+using T = int;
+using DV = dr::mhp::distributed_vector<T, dr::mhp::default_allocator<T>>;
+using V = std::vector<T>;
+
+TEST(MhpSort, Sort) {
+  std::vector<std::size_t> sizes = {
+      1, comm_size - 1, (comm_size - 1) * (comm_size - 1), 4, 7, 10, 23};
+
+  for (std::size_t n : sizes) {
+    std::cout << "Sort: Test size " << n << std::endl;
+    V l_v = generate_random<T>(n, 100);
+    DV d_v(n);
+    for (std::size_t idx = 0; idx < n; idx++) {
+      d_v[idx] = l_v[idx];
+    }
+    barrier();
+
+    std::sort(l_v.begin(), l_v.end());
+    dr::mhp::sort(d_v);
+
+    barrier();
+    EXPECT_TRUE(equal(l_v, d_v));
+  }
+}
+
+TEST(MhpSort, Sort_reverse) {
+  std::vector<std::size_t> sizes = {
+      1, comm_size - 1, (comm_size - 1) * (comm_size - 1), 4, 7, 10, 23};
+
+  for (std::size_t n : sizes) {
+    std::cout << "Sort_reverse: Test size " << n << std::endl;
+    V l_v = generate_random<T>(n, 1000);
+    DV d_v(n);
+
+    for (std::size_t idx = 0; idx < n; idx++) {
+      d_v[idx] = l_v[idx];
+    }
+    dr::mhp::barrier();
+
+    dr::mhp::sort(d_v, std::greater<T>());
+    std::sort(l_v.begin(), l_v.end(), std::greater<T>());
+    barrier();
+
+    EXPECT_TRUE(equal(l_v, d_v));
+  }
+}
```

Comment:
I think non-random sort tests are also needed. Please put them into fixed 3-ranks suite and create tests with predefined input, e.g:
[1,1,1,1,1,1,1,99,1,1,1] - many_small_values_one_greater
[10,15,13,19,  55, 53, 59, 51,  33, 31, 39, 34] - three_distinct_groups_test
[1,2,3,4,5,6] - sorted_input
[6,5,4,3,2,1] - reversed_sorted_input
[1,2,1,2,1,2,1,2,1,2,1,2] - ....
[1] - just one value
[] - no values
[1,2] - two values
[3,1,2] - three values
[3,1,2,4] - four values
float type test
[1,1,1,1,2,2,2,2] - sorted_two_val_only
[2,2,2,2,1,1,1,1] - reverse_sorted_two_val_only
[1,2,3,4,5,4,3,2,1] - pyramid_test
[5,4,3,2,1,2,3,4,5] - pyramid_reverse_test
[0,0,0,0,0,0,0,0,0] - same_value_test
[1, 2,... 1234567] million_sorted_test
[1234567, 1234566, ... 1] million_reverse_sorted_test
[...] big random test, simply: `{ test_sort_randomly(9876543, std::less<T>()); }`

Any other test cases which you think sort algorithm goes into corner cases, works badly, handles some rare `if`-s...

Reply:
0 to 4 elements tests are already in random tests (tests of size comm_size - 1 ). I think it is enough.
The other tests are added.